File: js/test/context.ts
import { Context } from './types';

export function initContext(): Context {
  const apiDomain = process.env.API_DOMAIN;
  if (!apiDomain) {
    throw Error('Missing env API_DOMAIN');
  }

  const apiKey = process.env.API_KEY;
  if (!apiKey) {
    throw Error('Missing env API_KEY');
  }

  const model = process.env.MODEL;
  if (!model) {
    throw Error('Missing env MODEL');
  }

  return {
    apiDomain,
    apiUrl: `https://${apiDomain}/v1`,
    apiKey,
    model,
  };
}

File: py/test/context.py
import os

from .types import Context


def init_context() -> Context:
    api_domain = os.getenv('API_DOMAIN')
    if not api_domain:
        raise ValueError('Missing env API_DOMAIN')

    api_key = os.getenv('API_KEY')
    if not api_key:
        raise ValueError('Missing env API_KEY')

    model = os.getenv('MODEL')
    if not model:
        raise ValueError('Missing env MODEL')

    return {
        'api_domain': api_domain,
        'api_url': f'https://{api_domain}/v1',
        'api_key': api_key,
        'model': model,
    }

